ethereum/statetransition/src/test/java/tech/pegasys/teku/statetransition/datacolumns/DataColumnSidecarCustodyImplTest.java

```diff
@@ -94,4 +95,72 @@ void sanityTest() throws Throwable {
     assertThat(fRet2_1.get().get()).isEqualTo(sidecar0);
     assertThat(fRet2_2.get().get()).isEqualTo(sidecar0);
   }
+
+  @Test
+  public void onSlot_shouldUpdateCustodyGroupCount() {
+    assertThat(custody.getTotalCustodyGroupCount()).isEqualTo(groupCount);
+    // Group count decreases
+    when(custodyGroupCountManager.getCustodyGroupCount()).thenReturn(groupCount - 2);
+    custody.onSlot(UInt64.ZERO);
+    assertThat(custody.getTotalCustodyGroupCount()).isEqualTo(groupCount - 2);
+    // Group count increases
+    when(custodyGroupCountManager.getCustodyGroupCount()).thenReturn(groupCount + 3);
+    custody.onSlot(UInt64.valueOf(config.getSlotsPerEpoch()).increment());
+    assertThat(custody.getTotalCustodyGroupCount()).isEqualTo(groupCount + 3);
+  }
+
+  @Test
+  public void onSlot_shouldUpdateFirstCustodyIncompleteSlotOnlyWhenCustodyGroupCountIncreases()
+      throws ExecutionException, InterruptedException {
+    // initial cgc is 0
+    custody =
+        new DataColumnSidecarCustodyImpl(
+            spec,
+            blockResolver,
+            dbAccessor,
+            minCustodyPeriodSlotCalculator,
+            custodyGroupCountManager,
+            0);
+    custody.onSlot(UInt64.ZERO);
+    // cgc increases to groupCount
+    when(custodyGroupCountManager.getCustodyGroupCount()).thenReturn(groupCount);
+    // first slot after which the min epoch for data column sidecars epoch is updated
+    final UInt64 firstSlotAfterMinEpochsForDataColumnSidecarsRequests =
+        UInt64.valueOf(config.getMinEpochsForDataColumnSidecarsRequests())
+            .increment()
+            .times(config.getSlotsPerEpoch());
+    custody.onSlot(firstSlotAfterMinEpochsForDataColumnSidecarsRequests);
+    final UInt64 advancedMinCustodyPeriodSlot =
+        minCustodyPeriodSlotCalculator.getMinCustodyPeriodSlot(
+            firstSlotAfterMinEpochsForDataColumnSidecarsRequests);
+    final UInt64 minCustodyPeriodSlot =
+        minCustodyPeriodSlotCalculator.getMinCustodyPeriodSlot(UInt64.ZERO);
+    assertThat(advancedMinCustodyPeriodSlot).isGreaterThan(minCustodyPeriodSlot);
+    assertThat(db.getFirstCustodyIncompleteSlot().get()).hasValue(advancedMinCustodyPeriodSlot);
+    // cgc stays the same
+    when(custodyGroupCountManager.getCustodyGroupCount()).thenReturn(groupCount);
+    // next epoch slot
+    final UInt64 firstEpochHopSlot =
+        firstSlotAfterMinEpochsForDataColumnSidecarsRequests
+            .plus(config.getSlotsPerEpoch())
+            .increment();
+    custody.onSlot(firstEpochHopSlot);
+    // min custody slot is not updated when cgc is the same
+    assertThat(db.getFirstCustodyIncompleteSlot().get()).hasValue(advancedMinCustodyPeriodSlot);
```

Comment:
You are not checking anything here because it was not ever advanced. Use mock for db instead:
```java
    final DataColumnSidecarDbAccessor dbAccessor = mock(DataColumnSidecarDbAccessor.class);
    when(dbAccessor.setFirstCustodyIncompleteSlot(any())).thenReturn(SafeFuture.COMPLETE);
```
and verify that there were only single interaction on cgc bump and no more interactions next